### DISCUSSION TITLE
how to access help files when lazy loading plugins?

### DISCUSSION THREAD

**Discussion by molleweide** [wbthomason/packer.nvim#918] [Q&A]
Hi there,

This feels like an embarrassing question, but how do I access help file to a plugin if I am lazy loading? 
I understand that I need to load somehow but how do I do this?

I am thinking do I need to write my own system for loading a plugin pre `:h plugingname` or is there a simple way that load a plugin in order to access a help tag?

#edit 

I went through the steps in #133 and learnt that some of my plugins have had their helptags generated, but not most of them. 
i need to do some more digging myself

**Comment by molleweide** [comment]
I found a related issue. however, I am not sure if it is the solution yet.
https://github.com/wbthomason/packer.nvim/issues/133

**Comment by blankname** [comment]
I'm using the [PackerComplete](https://github.com/wbthomason/packer.nvim#user-autocommands) `User` autocommand to copy the docs for opt plugins to a cache folder on my runtimepath and generating helptags there.
**Note:** I'm using the `vim.fs` module, which was only [merged](https://github.com/neovim/neovim/pull/18583) 4 days ago.


```
local opt_docs = vim.fn.stdpath('cache') .. '/docs/'
vim.opt.runtimepath:append(opt_docs .. '*')

vim.api.nvim_create_autocmd(
  "User",
  {
    pattern = 'PackerComplete',
    callback = function()
        local packer = require('packer')
        local opt_root = packer.config.package_root .. '/packer/opt/'
        local shellescape = vim.fn.shellescape

        -- Remove old doc directories/files.
        if vim.fn.isdirectory(opt_docs) == 1 then
          for name, _ in vim.fs.dir(opt_docs) do
            vim.fn.system('rm -r ' .. shellescape(opt_docs .. name))
          end
        end

        -- Copy documentation to the opt_docs directory and generate helptags
        for name, _type in vim.fs.dir(opt_root) do
          if _type == 'directory' then
            local doc_dir = name .. '/doc'
            local src =  opt_root .. doc_dir
            local dst = opt_docs .. doc_dir
            if vim.fn.isdirectory(src) == 1 then
              vim.fn.system('mkdir -p ' .. shellescape(vim.fs.dirname(dst)))
              vim.fn.system('cp -r ' .. shellescape(src) .. ' ' .. shellescape(dst))
              vim.cmd('helptags ' .. dst)
            end
          end
        end

        -- Trigger a runtimepath update.
        vim.opt.runtimepath = vim.opt.runtimepath
    end,
  }
)
```

**Reply by molleweide** [reply]
Thank you very much! Ima check this out asap.
Your solution got lost in my notifications.
